### ISSUE TITLE
[opentitan-tool] cw310 tests in bazel sometimes receive a truncated test rom and crash

### ISSUE THREAD

**Issue by drewmacrae** [lowRISC/opentitan#12739]
About 1% of the time, the test rom check reports:

```
[2022-05-18T14:52:04Z INFO  opentitanlib::util::rom_detect] Found: "TestROM:"
Error: cannot parse integer from empty string

Stack backtrace:
   0: <core::result::Result<T,F> as core::ops::try_trait::FromResidual<core::result::Result<core::convert::Infallible,E>>>::from_residual
   1: opentitanlib::util::rom_detect::RomDetect::detect
   2: <opentitanlib::transport::cw310::CW310 as opentitanlib::transport::Transport>::dispatch
   3: opentitanlib::app::TransportWrapper::dispatch
   4: <opentitantool::command::load_bitstream::LoadBitstream as opentitanlib::app::command::CommandDispatch>::run
   5: opentitantool::_::<impl opentitanlib::app::command::CommandDispatch for opentitantool::RootCommandHierarchy>::run
   6: opentitantool::execute
   7: opentitantool::main
   8: core::ops::function::FnOnce::call_once
   9: std::sys_common::backtrace::__rust_begin_short_backtrace
  10: std::rt::lang_start::{{closure}}
  11: core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &F>::call_once
             at /rustc/02072b482a8b5357f7fb5e5637444ae30e423c40/library/core/src/ops/function.rs:259:13
```

and the test fails. We should consider the following

- [ ] Debug the Test ROM reads to make them more consistent
- [x] When an empty Test ROM identifier is received we should just reload the Test ROM.

**Comment by drewmacrae** [comment]
For now I've marked the cw310 tests as flaky and bazel can work around the failures at very low cost.

**Comment by drewmacrae** [comment]
Can be reproduced with a command like:
```
bazel test --test_tag_filters=cw310,-broken --build_tests_only //sw/... --runs_per_test=10 --runs_per_test_detects_flakes
```

**Comment by drewmacrae** [comment]
repeating singular tests individually appears not to cause the failures [as often?]

**Comment by drewmacrae** [comment]
Thanks for your help with fixing opentitan-tool Chris. Seems to have worked and is included in #12493

**Comment by alphan** [comment]
Can we close this issue?
